src/LuaAST.ts

```diff
@@ -140,17 +125,25 @@ export function setParent(node: Node | Node[] | undefined, parent: Node): void
     if (Array.isArray(node)) {
         node.forEach(n => {
             n.parent = parent;
-            if (!n.line || !n.column) {
-                n.line = parent.line;
-                n.column = parent.column;
-            }
         });
     } else {
         node.parent = parent;
-        if (!node.line || !node.column) {
-            node.line = parent.line;
-            node.column = parent.column;
-        }
+    }
+}
+
+export function getOriginalPos(node: Node): { line: number, column: number } {
+    while (node.tsOriginal === undefined && node.parent !== undefined) {
+        node = node.parent;
+    }
+
+    if (node.tsOriginal !== undefined && node.tsOriginal.getSourceFile() !== undefined && node.tsOriginal.pos >= 0) {
```

Comment:
I had a TODO in the original PR. it should be investigated why node.tsOriginal.getSourceFile() can return undefined instead of catching that case.

Reply:
From what I could see working on this, it is probably because we create some ts nodes ourself in the transformer.